Lib/test/test_exceptions.py

```diff
@@ -1330,6 +1331,38 @@ def test_assert_shadowing(self):
             del AssertionError
             self.fail('Expected exception')
 
+    def test_memory_error_subclasses(self):
+        # MemoryError instances use a freelist of objects that are
+        # linked using the 'dict' attribute when they are
+        # inactive/dead. Subclasses of MemoryError should not
+        # participate in the freelist schema. This test creates a
+        # MemoryError object and keeps it alive (therefore
+        # advancing the freelist) and then it creates and destroys
+        # a subclass object. Finally, it checks that creating a
+        # new MemoryError succeeds, proving that the freelist is
+        # not corrupted. See bpo-41654 for more information.
+
+        class TestException(MemoryError):
+            pass
+
+        try:
+            raise MemoryError
+        except MemoryError as exc:
+            inst = exc
+
+        try:
+            raise TestException
+        except Exception:
+            pass
+
+        for _ in range(10):
+            try:
+                raise MemoryError
+            except MemoryError as exc:
+                pass
+
+            gc.collect()
```

Comment:
I'm not sure if it's still relevant in 2020, but there is support.gc_collect() which you may want to use here.